lib/chef/provider/windows_task.rb

```diff
@@ -0,0 +1,396 @@
+            Chef::Log.info "#{@new_resource} task ran"
+          end
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exists - nothing to do"
+        end
+      end
+
+      def action_delete
+        if @current_resource.exists
+          # always need to force deletion
+          run_schtasks 'DELETE', 'F' => ''
+          new_resource.updated_by_last_action true
+          Chef::Log.info "#{@new_resource} task deleted"
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exists - nothing to do"
+        end
+      end
+
+      def action_end
+        if @current_resource.exists
+          if @current_resource.status != :running
+            Chef::Log.debug "#{@new_resource} is not running - nothing to do"
+          else
+            run_schtasks 'END'
+            @new_resource.updated_by_last_action true
+            Chef::Log.info "#{@new_resource} task ended"
+          end
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exist - nothing to do"
+        end
+      end
+
+      def action_enable
+        if @current_resource.exists
+          if @current_resource.enabled
+            Chef::Log.debug "#{@new_resource} already enabled - nothing to do"
+          else
+            run_schtasks 'CHANGE', 'ENABLE' => ''
+            @new_resource.updated_by_last_action true
+            Chef::Log.info "#{@new_resource} task enabled"
+          end
+        else
+          Chef::Log.fatal "#{@new_resource} task doesn't exist - nothing to do"
+          raise Errno::ENOENT, "#{@new_resource}: task does not exist, cannot enable"
+        end
+      end
+
+      def action_disable
+        if @current_resource.exists
+          if @current_resource.enabled
+            run_schtasks 'CHANGE', 'DISABLE' => ''
+            @new_resource.updated_by_last_action true
+            Chef::Log.info "#{@new_resource} task disabled"
+          else
+            Chef::Log.warn "#{@new_resource} already disabled - nothing to do"
+          end
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exist - nothing to do"
+        end
+      end
+
+      private
+
+      # rubocop:disable Style/StringLiteralsInInterpolation
+      def run_schtasks(task_action, options = {})
+        cmd = "schtasks /#{task_action} /TN \"#{@new_resource.task_name}\" "
+        options.keys.each do |option|
+          cmd += "/#{option} "
+          cmd += "\"#{options[option].to_s.gsub('"', "\\\"")}\" " unless options[option] == ''
+        end
+        Chef::Log.debug('running: ')
+        Chef::Log.debug("    #{cmd}")
+        shell_out!(cmd, returns: [0])
+      end
+      # rubocop:enable Style/StringLiteralsInInterpolation
+
+      def task_need_update?
+        # gsub needed as schtasks converts single quotes to double quotes on creation
+        @current_resource.command != @new_resource.command.tr("'", '"') ||
+        @current_resource.user != @new_resource.user ||

```

Comment:
@NimishaS you shouldn't remove this line because, as you already said, without it we can't check for idempotency for `user` property.
```ruby
load_current_value do |desired|
  pathed_task_name = desired.task_name.start_with?('\\') ? desired.task_name : "\\#{desired.task_name}"

  task_hash = load_task_hash pathed_task_name

  task_name pathed_task_name
  if task_hash.respond_to?(:[]) && task_hash[:TaskName] == pathed_task_name
    exists true
    status :running if task_hash[:Status] == 'Running'
    enabled task_hash[:ScheduledTaskState] == 'Enabled' ? true : false
    cwd task_hash[:StartIn] unless task_hash[:StartIn] == 'N/A'
    command task_hash[:TaskToRun]
    user task_hash[:RunAsUser]
  else
    exists false
  end
end
```

Here we load current `user task_hash[:RunAsUser]` value based on output of 
```ruby
output = shell_out("schtasks /Query /FO LIST /V /TN \"#{task_name}\"").stdout
```

But the problem is that this output on russian systems will return `?????` instead of actual user, for example, `СИСТЕМА`. 
I think this is `shell_out` issue with codepage 866.

If I hardcoded `user` in `load_current_value do |desired|`
```ruby
load_current_value do |desired|
  pathed_task_name = desired.task_name.start_with?('\\') ? desired.task_name : "\\#{desired.task_name}"

  task_hash = load_task_hash pathed_task_name

  task_name pathed_task_name
  if task_hash.respond_to?(:[]) && task_hash[:TaskName] == pathed_task_name
    exists true
    status :running if task_hash[:Status] == 'Running'
    enabled task_hash[:ScheduledTaskState] == 'Enabled' ? true : false
    cwd task_hash[:StartIn] unless task_hash[:StartIn] == 'N/A'
    command task_hash[:TaskToRun]
    # user task_hash[:RunAsUser]
    user 'пользователь'
  else
    exists false
  end
end
```
IT WORKS AS EXPECTED, so the resource is idempotent.

Here is actual recipe
```ruby
windows_task 'task_from_name' do
  command 'dir'
  # user 'vagrant'
  # password 'vagrant'
  user 'пользователь'
  password 'Ohn9phev'
  run_level :highest
  action :create
end
```